R/zchunk_LA122.gasproc_refining.R

```diff
@@ -42,100 +42,370 @@ module_energy_LA122.gasproc_refining_DISABLED <- function(command, ...) {
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "out_gtl" | sector == "out_ctl") %>%
+      mutate(sector = if_else(sector=="out_gtl", "gtl", "ctl")) -> L122.out_EJ_R_gtlctl_Yh
+
+    # GTL and CTL inputs (L122.in_EJ_R_gtlctl_F_Yh): derived as output times exogenous input-output coefficients
+    # Interpolate gas processing IO coefs to all historical years and match in the fuel name
+    L122.globaltech_coef %>%
+      filter(sector == "gtl"| sector == "ctl") -> L122.gtlctl_coef
+
+    L122.out_EJ_R_gtlctl_Yh %>%
+      rename(valueInput = value) %>%
+      left_join(L122.gtlctl_coef, by = c("sector", "fuel", "year")) %>%
+      mutate(value_out = valueInput * value) %>%
+      select(-value, -valueInput) %>%
+      rename(value = value_out) %>%
+      select(GCAM_region_ID, sector, fuel, year, value) -> L122.in_EJ_R_gtlctl_F_Yh
+
+    # CRUDE OIL REFINING
+    # Copied from original text:
+    # NOTE*1: This is complicated. The outputs of CTL and GTL have the same fuel name as the output of oil refining,
+    # so need to be deducted from TPES in order to calculate regional output of oil refining.
+    # In contrast, biofuels are assigned different names, so they are not in the TPES of refined liquids.
+
+    #creating tibble with appropiate sectir and fuels for oil refining (output) for L122.out_EJ_R_oilrefining_Yh and adding historical years
+    tibble(GCAM_region_ID = GCAM_region_names$GCAM_region_ID, sector = "oil refining", fuel = "oil" ) -> L122.out_EJ_R_oilrefining_Yh
+    L122.out_EJ_R_oilrefining_Yh %>%
+      repeat_add_columns(tibble::tibble(year = HISTORICAL_YEARS)) -> L122.out_EJ_R_oilrefining_Yh
+
+    # Creating en_bal_TPES_OIL, en_bal_oil, ctl_OIL, and gtlctl_oil to adjust the outputs of CTL and GTL given the same fuel names of the oil refininf outputs (as mentioned in the note above)
+    # Getting output for refined liquids for oil refining (TPES) sector
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "TPES") %>%
+      filter(fuel == "refined liquids") %>%
+      select(GCAM_region_ID,sector, year, value_en_bal_TPES = value) %>%
+      mutate(sector = "oil refining") -> en_bal_TPES_OIL
+
+    # Getting output for refined liquids for net_oil refining sector
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "net_oil refining") %>%
+      filter(fuel == "refined liquids") %>%
+      select(GCAM_region_ID, sector, year, value_en_bal_net_oil = value) %>%
+      mutate(sector = "oil refining") %>%
+      left_join(en_bal_TPES_OIL, by = c("GCAM_region_ID", "sector", "year")) %>%
+      mutate(value_en_bal = value_en_bal_TPES - value_en_bal_net_oil) %>%
+      select(-value_en_bal_TPES, -value_en_bal_net_oil) -> en_bal_oil
+
+    # Getting output for coal for CTL sector
+    L122.out_EJ_R_gtlctl_Yh %>%
+      filter(sector == "ctl") %>%
+      filter(fuel == "coal") %>%
+      select(GCAM_region_ID, sector, year, value_ctl_oil = value) %>%
```

Comment:
All of these double `filter`s can be combined into `filter(x == ..., y = ...)` which might be clearer